mlx/linalg.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2023 Apple Inc.
+
+#include <iostream>
+#include <numeric>
+#include <set>
+#include <variant>
+
+#include "mlx/array.h"
+#include "mlx/linalg.h"
+#include "mlx/ops.h"
+
+namespace mlx::core::linalg {
+
+array vector_norm(
+    const array& a,
+    const std::variant<double, std::string>& ord,
+    const std::vector<int>& axes,
+    bool keepdims,
+    StreamOrDevice s) {
+  return std::visit(
+      overloaded{
+          [&](double p) {
+            if (p >= 1)
+              return power(
+                  sum(power(abs(a, s), array(p), s), axes, keepdims, s),
```

Comment:
I would special case `p==2` here because you can do `sqrt(sum(square(a, s)))` it will be faster and it's by far the most common case.

Reply:
Same comment for `p==1` special case `sum(abs(...`

Reply:
I agree. Thanks for pointing this out. I will implement as you suggested.